src/cloud-api-adaptor/cmd/process-user-data/main.go

```diff
@@ -37,7 +35,7 @@ var rootCmd = &cobra.Command{
 }
 
 func init() {
-	var agentConfigPath, cdhConfigPath, daemonConfigPath string
+	var cdhConfigPath, daemonConfigPath string
```

Comment:
we have to add `aaConfigPath`. process-user-data will not provision any files, but only a selected list of filenames set in `userdata.NewConfig()`

Reply:
@mkulke looks userdata.NewConfig does not have `aaConfigPath` here  https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/pkg/userdata/provision.go#L33, `aaConfigPath` is used only by `updateAgentConfigCmd` in https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/cmd/process-user-data/main.go#L59?

Reply:
yes, I think we need this.

```patch
From eb2e5174b01bdba299a792a70b004082a44b0f50 Mon Sep 17 00:00:00 2001
From: Magnus Kulke <magnuskulke@microsoft.com>
Date: Mon, 24 Jun 2024 16:13:39 +0200
Subject: [PATCH] process-user-data: ohai!

Signed-off-by: Magnus Kulke <magnuskulke@microsoft.com>
---
 src/cloud-api-adaptor/cmd/process-user-data/main.go |  6 ++++--
 src/cloud-api-adaptor/pkg/userdata/provision.go     | 11 +++++++++--
 2 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/src/cloud-api-adaptor/cmd/process-user-data/main.go b/src/cloud-api-adaptor/cmd/process-user-data/main.go
index d5ac81e..477c133 100644
--- a/src/cloud-api-adaptor/cmd/process-user-data/main.go
+++ b/src/cloud-api-adaptor/cmd/process-user-data/main.go
@@ -7,6 +7,7 @@ import (
 	"os"
 
 	cmdUtil "github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/cmd"
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/aa"
 	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/cdh"
 	daemon "github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/forwarder"
 	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/userdata"
@@ -35,18 +36,19 @@ var rootCmd = &cobra.Command{
 }
 
 func init() {
-	var cdhConfigPath, daemonConfigPath string
+	var aaConfigPath, cdhConfigPath, daemonConfigPath string
 	var fetchTimeout int
 
 	rootCmd.PersistentFlags().BoolVarP(&versionFlag, "version", "v", false, "Print the version")
 	rootCmd.PersistentFlags().StringVarP(&daemonConfigPath, "daemon-config-path", "d", daemon.DefaultConfigPath, "Path to a daemon config file")
+	rootCmd.PersistentFlags().StringVarP(&aaConfigPath, "aa-config-path", "a", aa.DefaultAaConfigPath, "Path to a AA config file")
 	rootCmd.PersistentFlags().StringVarP(&cdhConfigPath, "cdh-config-path", "c", cdh.ConfigFilePath, "Path to a CDH config file")
 
 	var provisionFilesCmd = &cobra.Command{
 		Use:   "provision-files",
 		Short: "Provision required files based on user data",
 		RunE: func(_ *cobra.Command, _ []string) error {
-			cfg := userdata.NewConfig(defaultAuthJsonPath, daemonConfigPath, cdhConfigPath, fetchTimeout)
+			cfg := userdata.NewConfig(aaConfigPath, defaultAuthJsonPath, daemonConfigPath, cdhConfigPath, fetchTimeout)
 			return userdata.ProvisionFiles(cfg)
 		},
 		SilenceUsage: true, // Silence usage on error
diff --git a/src/cloud-api-adaptor/pkg/userdata/provision.go b/src/cloud-api-adaptor/pkg/userdata/provision.go
index 8a045b2..df56a88 100644
--- a/src/cloud-api-adaptor/pkg/userdata/provision.go
+++ b/src/cloud-api-adaptor/pkg/userdata/provision.go
@@ -20,6 +20,7 @@ import (
 var logger = log.New(log.Writer(), "[userdata/provision] ", log.LstdFlags|log.Lmsgprefix)
 
 type paths struct {
+	aaConfig     string
 	authJson     string
 	cdhConfig    string
 	daemonConfig string
@@ -30,8 +31,8 @@ type Config struct {
 	paths        paths
 }
 
-func NewConfig(authJsonPath, daemonConfigPath, cdhConfig string, fetchTimeout int) *Config {
-	cfgPaths := paths{authJsonPath, cdhConfig, daemonConfigPath}
+func NewConfig(aaConfigPath, authJsonPath, daemonConfigPath, cdhConfig string, fetchTimeout int) *Config {
+	cfgPaths := paths{aaConfigPath, authJsonPath, cdhConfig, daemonConfigPath}
 	return &Config{fetchTimeout, cfgPaths}
 }
 
@@ -187,6 +188,12 @@ func processCloudConfig(cfg *Config, cc *CloudConfig) error {
 		return fmt.Errorf("failed to write daemon config file: %w", err)
 	}
 
+	if bytes := findConfigEntry(cfg.paths.aaConfig, cc); bytes != nil {
+		if err = writeFile(cfg.paths.aaConfig, bytes); err != nil {
+			return fmt.Errorf("failed to write aa config file: %w", err)
+		}
+	}
+
 	if bytes := findConfigEntry(cfg.paths.cdhConfig, cc); bytes != nil {
 		if err = writeFile(cfg.paths.cdhConfig, bytes); err != nil {
 			return fmt.Errorf("failed to write cdh config file: %w", err)
-- 
2.34.1
```

Reply:
Thanks @mkulke , applied the patch https://github.com/confidential-containers/cloud-api-adaptor/pull/1868/commits/7aacd90447c66db23ef6ba842dd5beb586814de7